packages/contracts/src/utils/JwtRegistry.sol

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.12;
+
+import "@zk-email/contracts/DKIMRegistry.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+import {MessageHashUtils} from "@openzeppelin/contracts/utils/cryptography/MessageHashUtils.sol";
+import {UUPSUpgradeable} from "@openzeppelin/contracts/proxy/utils/UUPSUpgradeable.sol";
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import { strings } from "solidity-stringutils/src/strings.sol";
+
+/// @title JWT Registry
+/// @notice TODO
+/// @dev TODO
+contract JwtRegistry is IDKIMRegistry, Ownable {
+
+    using strings for *;
+
+    DKIMRegistry public dkimRegistry;
+
+    // Check if azp is registered
+    mapping(string => bool) public whitelistedClients;
+
+    constructor(address _owner) Ownable(_owner) {
+        dkimRegistry = new DKIMRegistry(address(this));
+    }
+
+    /// @notice Checks if a public key hash is valid and not revoked for a given kis and iss.
+    /// @param domainName The domain name contains kis, iss and azp fields.
+    /// @param publicKeyHash The public key hash to validate.
+    /// @return bool Returns true if the public key hash is valid and not revoked, false otherwise.
+    function isDKIMPublicKeyHashValid(
+        string memory domainName,
+        bytes32 publicKeyHash
+    ) public view returns (bool) {
+        string[] memory parts = this.stringToArray(domainName);
+        string memory kidAndIss = string(abi.encode(parts[0], "|", parts[1]));
+        return dkimRegistry.isDKIMPublicKeyHashValid(kidAndIss, publicKeyHash) 
+          && whitelistedClients[parts[2]];
+    }
+
+    /// @notice Sets a public key hash for a `kis|iss` string  after validating the provided signature.
+    /// @param domainName The domain name contains kis, iss and azp fields.
+    /// @param publicKeyHash The public key hash to set.
+    /// @dev This function requires that the public key hash is not already set or revoked.
+    function setDKIMPublicKeyHash(
+        string memory domainName,
+        bytes32 publicKeyHash
+    ) public {
+        require(bytes(domainName).length != 0, "Invalid domain name");
+        require(publicKeyHash != bytes32(0), "Invalid public key hash");
+        string[] memory parts = this.stringToArray(domainName);
+        string memory kidAndIss = string(abi.encode(parts[0], "|", parts[1]));
+        require(
+            isDKIMPublicKeyHashValid(domainName, publicKeyHash) == false,
+            "publicKeyHash is already set"
+        );
+        require(
+            dkimRegistry.revokedDKIMPublicKeyHashes(publicKeyHash) == false,
+            "publicKeyHash is revoked"
+        );
+
+        dkimRegistry.setDKIMPublicKeyHash(kidAndIss, publicKeyHash);
+        // Register azp
```

Comment:
Can you please not use DKIM as we are not using it?

Reply:
@shreyas-londhe I am doing this to match the logic with using the previous registry. does this mean I should store the public key hash in a different way?